wp_mobile_cache/src/mappings/posts.rs

```diff
@@ -0,0 +1,333 @@
+            },
+            content: PostContentWithEditContext {
+                raw: row.get_column(ContentRaw)?,
+                rendered: row.get_column(ContentRendered)?,
+                protected: row.get_column(ContentProtected)?,
+                block_version: row.get_column(ContentBlockVersion)?,
+            },
+            author: get_optional_id(row, Author)?,
+            excerpt: {
+                // Presence of excerpt is determined by excerpt_rendered being Some
+                let excerpt_rendered: Option<String> = row.get_column(ExcerptRendered)?;
+                if excerpt_rendered.is_some() {
+                    Some(SparsePostExcerpt {
+                        raw: row.get_column(ExcerptRaw)?,
+                        rendered: excerpt_rendered,
+                        protected: row.get_column(ExcerptProtected)?,
+                    })
+                } else {
+                    None
+                }
+            },
+            featured_media: get_optional_id(row, FeaturedMedia)?,
+            comment_status: parse_optional_enum(row, CommentStatus)?,
+            ping_status: parse_optional_enum(row, PingStatus)?,
+            format: parse_optional_enum(row, Format)?,
+            meta: deserialize_json_value(row.get_column(Meta)?)?,
+            sticky: integer_to_bool(row.get_column(Sticky)?),
+            template: row.get_column(Template)?,
+            categories: None,
+            tags: None,
+            parent: get_optional_id(row, Parent)?,
+            menu_order: row.get_column(MenuOrder)?,
+        };
+
+        Ok(Self {
+            row_id,
+            site,
+            post,
+            last_fetched_at: row.get_column(LastFetchedAt)?,
+        })
+    }
+}
+
+impl DbEntity for AnyPostWithEditContext {
+    const TABLE_NAME: &'static str = "posts_edit_context";
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        DbSite,
+        repository::posts::PostRepository,
+        test_fixtures::posts::{create_full_post, create_minimal_post},
+        test_helpers::{test_db, test_site},
+    };
+    use rstest::*;
+    use rusqlite::Connection;
+    use wp_api::posts::{PostId, PostStatus};
+
+    /// Helper to validate that last_fetched_at is a recent, valid ISO 8601 timestamp
+    fn assert_recent_timestamp(timestamp: &str) {
+        // Parse the timestamp
+        assert!(
+            timestamp.ends_with('Z'),
+            "Timestamp should be UTC (end with Z): {}",
+            timestamp
+        );
+        assert!(
+            timestamp.contains('T'),
+            "Timestamp should be ISO 8601 format: {}",
+            timestamp
+        );
+        // Basic format check: YYYY-MM-DDTHH:MM:SS.fffZ
+        assert!(
+            timestamp.len() >= 20,
+            "Timestamp should be at least 20 chars: {}",
+            timestamp
+        );
+    }
+
+    #[rstest]
+    fn test_round_trip_with_minimal_fields(mut test_db: Connection, test_site: DbSite) {
```

Comment:
We can combine these two tests into cases with minimal and full as they test the same thing.